### DISCUSSION TITLE
Extending React 18 transition support

### DISCUSSION THREAD

**Discussion by brncsk** [facebookexperimental/Recoil#1823] [Ideas]
Hi all,
thank you for Recoil, it's arguably my favorite piece of software nowadays.

I've started experimenting with React 18 transition support lately, which works well afaict, but the API surface feels lacking. Do you folks plan on expanding it in the future? (Namely, `useRecoilCallback` and `useResetRecoilState` feel like two worthy candidates to have their `_TRANSITION_SUPPORT_UNSTABLE` variants).

Also, the Recoil docs mention the following:

> However, React 18 may fallback from concurrent updates and does not yet officially support initiating transitions based on state changes to external stores. This is something the React team is looking into supporting, but until then we have added experimental support for this through the following hooks.

I'm not familiar with the priorities of the React team on this – is there any GitHub issue/discussion where I could track their progress on supporting this? Thank you.

**Comment by drarmstr** [comment]
`useResetRecoilState()` actually doesn't need a `_TRANSITION_SUPPORT_UNSTABLE` variant as the difference is actually only when reading the state, not setting it.
